src/App.tsx

```diff
@@ -7,33 +9,78 @@ function getRandomName(): string {
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type Props = {};
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+type State = {
+  time: string;
+  oldName: string,
+  newName: string,
+  hasClock: boolean,
+};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    time: '',
+    oldName: 'Clock-0',
+    newName: '',
+    hasClock: true,
+  };
+
+  componentDidMount() {
+    window.setInterval(() => {
+      this.setState({ time: new Date().toLocaleTimeString() });
+    }, 1000);
+
+    window.setInterval(() => {
+      if (this.state.hasClock === true) {
+        console.info(this.state.time);
+      }
+    }, 1000);
+
+    window.setInterval(() => {
+      this.setState({ newName: getRandomName() });
+
+      if (this.state.hasClock === true) {
+        console.info(`Renamed from ${this.prevState()} to ${this.state.newName}`);
```

Comment:
try to use `componentDidUpdate()` method to log this message
you can get `prevState` from this method parameters